internal/pkg/deploy/deploy.go

```diff
@@ -59,11 +59,7 @@ type Store struct {
 }
 
 // NewStore returns a new store.
-func NewStore() (*Store, error) {
-	store, err := config.NewStore()
-	if err != nil {
-		return nil, fmt.Errorf("connect to config store: %w", err)
-	}
+func NewStore(store configStoreClient) (*Store, error) {
```

Comment:
out-of-curiosity what does it look like for a client that wants to use the constructor if the interface is private?
What does the error message look like if they pass an obj that doesnt implement the interface :O or what do the docs look like

Reply:
> out-of-curiosity what does it look like for a client that wants to use the constructor if the interface is private?

https://github.com/aws/copilot-cli/pull/1090/files#diff-88fdaab0450f79a9400f9ba9c6d7c280R39

> What does the error message look like if they pass an obj that doesnt implement the interface :O or what do the docs look like

it will not be compilable i think...